### PR TITLE
[uriparser] Update to v0.9.5

### LINKED ISSUE #19905: [uriparser] update to 0.9.5
Library name: uriparser

New version number: 0.9.5

Other information that may be useful (release notes, etc...)
https://github.com/uriparser/uriparser/releases

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- #### What does your PR fix?  
  - Update to v0.9.5
  - CONTROL -> vcpkg.json
  - Modernize

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@myd7349, thanks for the PR! Please let me know if the PR ready for review.

File: ports/uriparser/portfile.cmake [hunk 1]
```diff
@@ -12,9 +12,8 @@ else()
     set(URIPARSER_BUILD_TOOLS OFF)
 endif()
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (PhoebeHui) [inline ports/uriparser/portfile.cmake:16]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```
```suggestion
    SOURCE_PATH ${SOURCE_PATH}
```

Comment (PhoebeHui) [inline ports/uriparser/portfile.cmake:16]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**